src/opm/output/eclipse/Summary.cpp

```diff
@@ -3059,9 +3086,18 @@ SummaryImplementation(const EclipseState&  es,
     , fmt_           { es.cfg().io().getFMTOUT() }
     , unif_          { es.cfg().io().getUNIFOUT() }
 {
+    const auto st = SummaryState {
```

Comment:
Hmmm - I have not really looked at this - but instantiating a `SummaryState`in here looks suspicious? That should be managed by calling scope?

Reply:
This is the constructor function for `Summary`&mdash;well technically for `SummaryImplementation`.  This constructor is called once, very early in the simulation run and before any wells or groups have become active so there's no interesting state to inspect.  I therefore don't think there's any benefit to using an object managed by a caller.  Moreover, this `SummaryState` object is `const` because we're not actually updating it here.  We merely need it to exist in order to call evaluation functions with an [empty set of wells](https://github.com/OPM/opm-common/blob/12dd9d321b11e3780527220daa220791e5021500/src/opm/output/eclipse/Summary.cpp#L2729) to derive the appropriate [unit string](https://github.com/OPM/opm-common/blob/12dd9d321b11e3780527220daa220791e5021500/src/opm/output/eclipse/Summary.cpp#L2735) for a summary variable.